### PR TITLE
add match filters

### PR DESCRIPTION
**Pull request by nmartin413**
Adds match filters for use in panther-okta.

### CODE REVIEW

**Comment by darwayne** [comment]
looking 👀

File: panther_utils/match_filters.py [hunk 1]
```diff
@@ -27,3 +27,53 @@ def _deep_equal(obj: typing.Any, params: typing.Any) -> bool:
...
+    )
+
+
+def _deep_equal_pattern(obj: typing.Any, params: typing.Any) -> bool:
+    import re
+    import functools
+    import collections
+
+    path = params["path"]
+    keys = path.split(".")
```
Comment (darwayne) [inline panther_utils/match_filters.py:45]: does this mean no json path support out of the box? will we have to circle back and add that?

Comment (nmartin413) [reply panther_utils/match_filters.py:45]: not initially, but the simple dot are a subset of json path, so we can add it later just by updating this repo.

**Review by darwayne — DISMISSED** [review]

**Comment by darwayne** [comment]
taking another look

**Review by darwayne — APPROVED** [review]

**Review by maxrichie5 — APPROVED** [review]

**Pull request merged**